cmd/load-generator/boulder-calls.go

```diff
@@ -0,0 +1,482 @@
+package main
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/sha1"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	mrand "math/rand"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/letsencrypt/boulder/core"
+
+	"gopkg.in/square/go-jose.v1"
+)
+
+// not entirely sure this will actually work...?
+var (
+	stringToOperation = map[string]func(*State, *context) error{
+		"newRegistration":   newRegistration,
+		"getRegistration":   getRegistration,
+		"newAuthorization":  newAuthorization,
+		"solveHTTPOne":      solveHTTPOne,
+		"solveTLSOne":       solveTLSOne,
+		"newCertificate":    newCertificate,
+		"revokeCertificate": revokeCertificate,
+	}
+)
+var plainReg = []byte(`{"resource":"new-reg"}`)
+
+func newRegistration(s *State, ctx *context) error {
+	// if we have generated the max number of registrations just become getRegistration
+	if s.maxRegs != 0 && s.numRegs() >= s.maxRegs {
+		return getRegistration(s, ctx)
+	}
+	// signKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	signKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return err
+	}
+	// signKey.Precompute()
+	signer, err := jose.NewSigner(jose.ES256, signKey)
+	if err != nil {
+		return err
+	}
+	ctx.ns = &nonceSource{s: s}
+	signer.SetNonceSource(ctx.ns)
+
+	// create the registration object
+	var regStr []byte
+	if s.email != "" {
+		regStr = []byte(fmt.Sprintf(`{"resource":"new-reg","contact":["mailto:%s"]}`, s.email))
+	} else {
+		regStr = plainReg
+	}
+	// build the JWS object
+	requestPayload, err := s.signWithNonce("/acme/new-reg", true, regStr, signer)
```

Comment:
It might be worthwhile making a `const` for the `/acme/new-reg` path components. There's some duplication throughout that could be collapsed.